packages/grid/x-data-grid/src/hooks/features/rows/useGridParamsApi.ts

```diff
@@ -154,6 +160,38 @@ export function useGridParamsApi(apiRef: React.MutableRefObject<GridPrivateApiCo
     [apiRef, getBaseCellParams],
   );
 
+  const getRowValue = React.useCallback<GridParamsApi['getRowValue']>(
+    (row, colDef) => {
+      const id = getRowId ? getRowId(row) : row.id;
```

Comment:
It seems to produce the same result as `getCellValue`. What is the added value here?

Reply:
This is related to point 9: if we're iterating the row objects directly, we need a way to know the cell value without going through `dataRowIdToModelLookup`, which `.getCellValue` does. Big lookup objects are expensive, in particular if we need to access them in a loop that's already `O(n)` to start with.

Reply:
Would using `Set` for the `dataRowIdToModelLookup` help here?

Reply:
\*`Map`

No, I've tried using Maps and it slows down the benchmark. The issue is really the indirection & additional memory accesses. The `row` object is basically a direct pointer to the row data. Passing `row.id` to an hashmap to get the row data again is wasted cycles.

Reply:
Is `dataRowIdToModelLookup` slow, because we change its shape many times (adding more and more keys to it) and therefore all its properties are "slow properties" (according to https://v8.dev/blog/fast-properties)?
Is this correct?

Reply:
Yeah exactly. Small & consistent objects are fast, big objects always end up in slow mode: the best the engines can do at some point is a hashmap.

I'm not sure why using a `Map` slows things down though. I would expect that it would speed things up, because the engine doesn't lose time trying to figure out the object shape, it's just a hashmap from the start.

Reply:
Gotcha. I have a suggestion for the `getCellValue` then - can we support conditional `row` and `column` arguments and use them if available, ani f not - it will use `getRow()` and `getColumn` as fallback.
What do you think?

Reply:
What would be the signature?

Reply:
I have a feeling that we'll always have either both objects or none. How about we keep the API like it's implemented but leave it undocumented, and we can keep iterating & refactoring it before v7? I'll open a PR for the v7 filters shortly, we can continue the discussion there.

Reply:
https://github.com/mui/mui-x/pull/9254